### ISSUE TITLE
Issue when i try analyze file with cuckoo!

### ISSUE THREAD

**Issue by manolete66** [cuckoosandbox/cuckoo#2547]
Thanks for creating an issue! But first: did you read our community guidelines?
https://cuckoo.sh/docs/introduction/community.html

##### My issue is:
Hellow, im a student who is trying to set cuckoo enviorement for a class project. I configured all following the community guidelines of cuckoo but i get a issue.

In this link i show what the running cuckoo take out for screen.
https://ibb.co/mCVptL

I need to point that i ran agent.py in the virtual machine (in my case windows7) as administrator and to transfer the data to analyze i opened a new terminal and wrote "cuckoo submit ejemplo" and this showed me a sucessfull submit message.

If someone can help me I will be grateful. Thanks for your time.

##### My Cuckoo version and operating system are:

Cuckoo Sandbox 2.0.6
Ubuntu 16.04 LTS

**Comment by doomedraven** [comment]
1. did you take snapshot in running state?
2. if 1 is yes, are you able to do `curl vm_ip:8000` once analysis start

**Comment by doomedraven** [comment]
is that error happens to all your samples or only some?

**Comment by manolete66** [comment]
Okay, ill try with other example to analyze. Last question, when i submit the task to analyze, in cuckoo logs i should see something which indicates that the task uploaded correctly?

**Comment by doomedraven** [comment]
read last 2 lines from your image here https://ibb.co/fkJ7Xf

**Comment by manolete66** [comment]
Hi again doomedraven, i got execute my own task (the file virusPrueba.bat appointed before) with cuckoo and this is the result:

https://ibb.co/faWqcq

Im not sure about the warning that cuckoo shows. Why no logs are generated?

This image contents the storage folder of the analisys:

https://ibb.co/bOimHq

Note: buffer, extracted, files, logs, network, shots are empty.

**Comment by doomedraven** [comment]
yep it means something went wrong, it can be what your script generate too much behavior so it need to be increased(enable -d for debug to see that msg) or idl, but run with debug mode so maybe there some clues

**Comment by manolete66** [comment]
Here u can see doom, the debug of cuckoo's execution:

https://ibb.co/jfCGkA
https://ibb.co/hqstXq

Idk what is the problem, its like all be good except for the showed warning.
Thanks for ur time.

**Comment by doomedraven** [comment]
can you post the vbox conf from configs ?

**Comment by manolete66** [comment]
virtualbox.conf:

https://ibb.co/huD8Xq
https://ibb.co/gVJxQA
https://ibb.co/h3n2sq

cuckoo.conf:

https://ibb.co/e7BvCq
https://ibb.co/j4F0dV
https://ibb.co/bFBTXq
https://ibb.co/nwPhsq

**Comment by doomedraven** [comment]
all looks fine weird, what about `ifconfig` output? + do  you have any custom iptables?

**Comment by manolete66** [comment]
ifconfig:
https://ibb.co/ksSoFA

I havent got configured ip tables, i need it for communicate host and guest?
I configured a only-host adapter and i can ping correctly between host and guest.

**Comment by doomedraven** [comment]
if i tell you truth im not sure what happens, you need to start tcpdump or wireshark and investigate why it has problem to send packages back, or wait

can you explain me step by step how did you run cuckoo? which users root and cuckoo? only cuckoo there could be a problem

**Comment by manolete66** [comment]
I run cuckoo with my own user whose name is "equipo".
I added it to vbox users with this command:

sudo usermod -a -G vboxusers cuckoo

I didnt do more for users configuration.

**Comment by manolete66** [comment]
Doom, i advanced. I configured ip tables and now i could do the analyze of the file from the web interface of cuckoo. Here u can see the results:

https://ibb.co/nOLE5A
https://ibb.co/d8YCsq
https://ibb.co/k6zVdV
https://ibb.co/egydXq
https://ibb.co/hn0gkA

Nevertheless, i got the same warning:

https://ibb.co/bP58Xq
I think that this execution was sucessfully but idk what is the warning and why it appears.
The same directorys which i said that were empty are empty again (in storage, analyze 5, buffer, extracted, files, logs and shots)

**Comment by doomedraven** [comment]
vale por partes :P
1. vboxusers y cuckoo son diferentes 
prueba ejecutar todo en diferentes terminales 
1. sudo cuckoo -d rooter -g equipo
2. cuckoo -d
3. cuckoo -d web -H 0.0.0.0
4. cuckoo -d process a1

**Comment by foolishhare** [comment]
> Doom, i advanced. I configured ip tables and now i could do the analyze of the file from the web interface of cuckoo. Here u can see the results:
> 
> https://ibb.co/nOLE5A
> https://ibb.co/d8YCsq
> https://ibb.co/k6zVdV
> https://ibb.co/egydXq
> https://ibb.co/hn0gkA
> 
> Nevertheless, i got the same warning:
> 
> https://ibb.co/bP58Xq
> I think that this execution was sucessfully but idk what is the warning and why it appears.
> The same directorys which i said that were empty are empty again (in storage, analyze 5, buffer, extracted, files, logs and shots)

1) How named your user in guest VM(Windows)?
2) Did you run agent.py with admin privileges?
I had same problem. My user in VM was cuckoo - so i changed it to "Administrator"
And add agent.py to Task Scheduler, and mark checkbox "Run with highest privileges "
That solve my problem.

**Comment by doomedraven** [comment]
My friend told me what this happens with latests versions of vbox only

**Comment by doomedraven** [comment]
Puedes probar version un pelin mas desqctualuzada de vbox pa ver

**Comment by manolete66** [comment]
The life in cuckoo is hard :(
I will try with a older version of VB and after i'll tell u the result.

**Comment by doomedraven** [comment]
vbox sucks :P use kvm https://github.com/doomedraven/Tools/blob/master/Virtualization/kvm-qemu.sh

**Comment by manolete66** [comment]
Im installing the graphic interface of Virtual-Manager and configuring the guest and network enviroment. When i finish i'll tell u about the results of a cuckoo analyze.

PD: Some tips for the configuration of guest/network-guest in Virtual-manager?

**Comment by doomedraven** [comment]
@manolete66 i hope you figurated that out no? it pretty easy to configure networking and everything with virt-manager

**Comment by doomedraven** [comment]
try also cuckoo -d -g equipo

**Comment by manolete66** [comment]
When i enter the above command i recive a error:

Error: no such option: -g

Idk if this command should work or whats happening.

**Comment by doomedraven** [comment]
`cuckoo --help`

**Comment by doomedraven** [comment]
ok --user equipo

**Comment by doomedraven** [comment]
That is bcz default group and user is cuckoo is why you should run
everything under that user and not custom, for custom you need to use - -
user and - g

Add cuckoo user change to it and enjoy

**Comment by manolete66** [comment]
I cant  run cuckoo -d --user equipo becouse it says: 

Failed to drop privileges to equipo: [Errno 1] Operation not permitted

Then it cant find the option -g: 

Error: no such option: -g

**Comment by manolete66** [comment]
Im a few lost in this permission problem.

**Comment by doomedraven** [comment]
--help helps
Add cuckoo user and run everything under it and issue will be solved

**Comment by manolete66** [comment]
okey i will try, thanks for all doomedraven.

**Comment by manolete66** [comment]
Ok doome, i did the next:

1) sudo adduser cuckoo
2) usermod -G libvirtd -a cuckoo
3) usermod -G kvm -a cuckoo
4) I closed session in my computer and enter again.
5) Ran cuckoo -d --user cuckoo
Got this error: 

2018-12-17 11:59:11,405 [cuckoo] CRITICAL: CuckooCriticalError: Error initializing machines: Cannot connect to libvirt

What is happening?

**Comment by doomedraven** [comment]
basically you can't connect to kvm/qemu, but you are in group, check in case if group name is libvirt(without d), i saw that once
